### PR TITLE
Create template folder

### LINKED ISSUE #98: [FEATURE REQUEST] README template
**Is your feature request related to a problem? Please describe.**
We would like the README.md in each dataset to have similar formats. 

**Describe the solution you'd like**
Add a template README.md.

### PR DESCRIPTION
**Pull request by xingjian-zhang**
## Description
Create a template folder for contributors' reference.

## Related Issue

To fix #98

## Motivation and Context

We want users can refer to a concise template when writing their configuration files. This should be ideally like a GitHub Pull Request/Issue template.

## How Has This Been Tested?

Not tested yet.

### CODE REVIEW

**Comment by xingjian-zhang** [comment]
I found a format called [hjson](https://hjson.github.io), which is essentially a json format with comments. I created a metadata.json template based on that. What do you think about it, @jiaqima?

**Comment by jiaqima** [comment]
> I found a format called [hjson](https://hjson.github.io), which is essentially a json format with comments. I created a metadata.json template based on that. What do you think about it, @jiaqima?

Sounds good.

File: examples/template/metadata.hjson [hunk 1]
```diff
@@ -0,0 +1,78 @@
+{
+  /*
+  This configuration is written in hjson format, which is essentially json with
+  comments. Please use this file as a template of a homogeneous graph's
+  metadata.json. See more about metadata format requirement at FORMAT.md. Note
+  that
+  - metadata.json is still a json file.
```
Comment (jiaqima) [inline examples/template/metadata.hjson:7]: ... a json file. All comments should be removed in json files.

File: examples/template/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+<!-- Replace the title with the dataset name. -->
+
+## Dataset Description
+
+<!-- A concise description of the dataset. -->
+
+#### Citation
+
+```
```
Comment (jiaqima) [inline examples/template/README.md:11]: Add the bullet points `Original Source`, `Current Version`, and `Previous Versions`, where the latter two are optional. See [cora](https://github.com/Graph-Learning-Benchmarks/gli/tree/main/datasets/cora) as an example.

File: examples/template/preprocess.ipynb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Please use this file to write codes that transform the data files and configuration files from the raw data. Typically this script should include following parts:\n",
```
Comment (jiaqima) [inline examples/template/preprocess.ipynb:7]: Maybe mention that it could also be a .py file in addition to .ipynb?

**Review by jiaqima — APPROVED** [review]
Left some minor comments. Otherwise LGTM.

File: examples/template/README.md (hunk 2 above)
Comment (xingjian-zhang) [reply examples/template/README.md:11]: Could you please give a definition of these versions?

Comment (jiaqima) [reply examples/template/README.md:11]: Original source refers to the first work that created the dataset. Current version refers to the work that is directly responsible for the exact form of the dataset stored in this folder. Previous versions refer to any intermediate versions between original source and the current version. There could be multiple citations for the previous versions.

**Comment by jiaqima** [comment]
Oh, I just found that those "Original source" etc were inserted in "Available Tasks" section. It should be inserted in the dataset description section.

**Comment by xingjian-zhang** [comment]
My bad. I will open another PR.

**Pull request merged**